### PR TITLE
Fix: `preferredSkinTone` type warning

### PR DESCRIPTION
**Pull request by tienifr**
### Details

The `preferredSkinTone` is supposed to be `Number` but sometimes it's passed as `String` since `default` was its legacy default value (becomes `-1` now). This PR fixes that.

### Fixed Issues

$ https://github.com/Expensify/App/issues/18541
PROPOSAL: https://github.com/Expensify/App/issues/18541#issuecomment-1543832625

### Tests

1. Login with any account
2. Go to any chat
3. Type in semicolon with emoji prefix (i.e. `:sm`)
4. Verify that the error _Invalid prop `preferredSkinToneIndex` of type `string` supplied to `EmojiSuggestions`, expected `number`_ does not appear in the JS console

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Login with any account
2. Go to any chat
3. Type in semicolon with emoji prefix (i.e. `:sm`)
4. Verify that the error _Invalid prop `preferredSkinToneIndex` of type `string` supplied to `EmojiSuggestions`, expected `number`_ does not appear in the JS console

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 12-05-2023 01:42:18.webm](https://github.com/Expensify/App/assets/113963320/aa6c941d-ed1e-4547-8d08-4785d09d3fc4)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

[Screencast from 12-05-2023 01:53:16.webm](https://github.com/Expensify/App/assets/113963320/759f70e2-9113-4cc9-8aaf-150667bca02e)

https://github.com/Expensify/App/assets/113963320/a32eefaa-9363-4b9a-a241-8c3b0f380481

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/113963320/6dc02ee6-0f90-4827-bffc-8ef66da9b914

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/113963320/8a338dcd-75ac-4134-bfcf-769bbc5ca08f

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/113963320/8f57c512-850c-434d-accf-dbe280f42cf7

</details>

<details>
<summary>Android</summary>

[untitled.webm](https://github.com/Expensify/App/assets/113963320/76f4b8ad-6cfd-4054-aa05-43a873271f3d)

</details>

### CODE REVIEW

**Comment by tienifr** [comment]
While working on this PR, I found out that if the new `getPreferredSkinToneIndex` function was put in `EmojiUtils` then using it in `Report.js` would cause import cycle: `Report` > `EmojiUtils` > `User` > `Report`. Thus, I decided to move this line out of `EmojiUtils`:

https://github.com/tienifr/App/blob/057417c49ce8f9d2834d2dcab6546523a277cbe6/src/libs/EmojiUtils.js#L191

due to:
1. Prone to causing import cycle
2. Calling API function within an utility file is weird

Then it would be used like this:
https://github.com/tienifr/App/blob/66f7c0ab8c64749b8d2e6a3ec2f7fe6357b38086/src/pages/home/report/ReportActionCompose.js#L610-L611

**Comment by pecanoro** [comment]
There are some unit tests failing!

File: src/libs/actions/Report.js
Comment (mollfpr) [inline src/libs/actions/Report.js:44]: @tienifr Is `preferredSkinTone` not taking any value from Onyx? `EmojiUtils.getPreferredSkinToneIndex` is returning a number, shouldn't we assign the value to `preferredSkinTone`?

**Comment by tienifr** [comment]
Comments're resolved and tests're fixed!

**Comment by pecanoro** [comment]
@mollfpr Waiting on the checklist to do a final review from my side 😄

**Comment by mollfpr** [comment]
Sorry I’ll finish it tonight!

File: src/libs/EmojiUtils.js
Comment (mollfpr) [inline src/libs/EmojiUtils.js:176]: I think the name `addToFrequentlyUsedEmojis` isn't valid anymore, we could rename it to `getFrequentlyUsedEmojis`.

**Review by mollfpr — CHANGES_REQUESTED** [review]

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



https://github.com/Expensify/App/assets/25520267/da216af8-17fb-4f29-b19a-87a1397e2ec1


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/25520267/40d8fbe1-da64-4e9f-8ade-fb0ce75af9bb


</details>

<details>
<summary>Mobile Web - Safari</summary>



https://github.com/Expensify/App/assets/25520267/35bb04dc-995b-435b-ab7b-0c7027e60836


</details>

<details>
<summary>Desktop</summary>



https://github.com/Expensify/App/assets/25520267/cf66cc81-d613-44e3-8342-9655e7c95518


</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/25520267/6525ec45-5157-443d-b0e9-14b6cbee127d


</details>

<details>
<summary>Android</summary>



https://github.com/Expensify/App/assets/25520267/8c53968a-1d80-419e-a82f-728da27ec2be


</details>

File: src/libs/EmojiUtils.js
Comment (mollfpr) [inline src/libs/EmojiUtils.js:258]: Should this update the frequently used emojis user?

Comment (tienifr) [reply src/libs/EmojiUtils.js]: Thanks for pointing that out. I've missed this function usage before.

I think that we also need to move this line out of `replaceEmojis` function and code a separate logic to handle the API update. Please take a look at my new code.

**Comment by pecanoro** [comment]
@tienifr Linter is complaining since you need to run `npm run prettier`!

**Review by pecanoro — APPROVED** [review]

**Pull request merged**
